File: leetcode/83.删除排序链表中的重复元素/83.删除排序链表中的重复元素.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

func deleteDuplicates(head *ListNode) *ListNode {
	if head == nil{
		return head
	}
	slow := head
	fast := head
	for fast != nil{
		if slow.Val != fast.Val{
			slow = slow.Next
			slow.Val = fast.Val
		}
		fast = fast.Next
	}
	slow.Next = nil
	return head
}

File: leetcode/83.删除排序链表中的重复元素/83.删除排序链表中的重复元素.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        if head is None:
            return head
        slow = fast = head
        while fast:
            if slow.val != fast.val:
                slow = slow.next
                slow.val = fast.val
            fast = fast.next
        # 断开与后面重复元素的连接
        slow.next = None
        return head
